qiskit_nature/mappers/second_quantization/logarithmic_mapper.py

```diff
@@ -0,0 +1,199 @@
+
+        # Embed the spin matrices in a larger matrix of size 2**num_qubits x 2**num_qubits
+        embed = lambda matrix: self._embed_matrix(
+            matrix,
+            num_qubits,
+            embed_padding=self._embed_padding,
+            embed_location=self._embed_location,
+        )
+        embedded_spin_matrices = list(map(embed, spin_matrices))
+
+        # Generate operators from these embedded spin matrices
+        embedded_operators = [Operator(matrix) for matrix in embedded_spin_matrices]
+        for op in embedded_operators:
+            op = SparsePauliOp.from_operator(op)
+            op.chop()
+            spin_op_encoding.append(PauliSumOp(1.0 * op))
+
+        return spin_op_encoding
+
+    def _embed_matrix(
+        self,
+        matrix: np.ndarray,
+        num_qubits: int,
+        embed_padding: float = 1,
+        embed_location: str = "upper",
+    ) -> np.ndarray:
+        """
+        Embeds `matrix` into the upper/lower diagonal block of a 2^num_qubits by 2^num_qubits matrix
+        and pads the diagonal of the upper left block matrix with the value of `embed_padding`.
+        Whether the upper/lower diagonal block is used depends on `embed_location`.
+        I.e. using embed_location = 'upper' returns the matrix:
+            [[ matrix,    0             ],
+            [   0   , embed_padding * I]]
+        Using embed_location = 'lower' returns the matrix:
+            [[ embed_padding * I,    0    ],
+            [      0           ,  matrix ]]
+        Args:
+            matrix: The matrix (2D-array) to embed.
+            num_qubits: The number of qubits on which the embedded matrix should act on.
+            embed_padding:
+                The value of the diagonal elements of the upper left block of the embedded matrix.
+            embed_location: Must be one of ['upper', 'lower']. This parameters sets
+                whether the given matrix is embedded in the
+                upper left hand corner or the lower right hand corner of the larger matrix.
+
+        Returns:
+            If `matrix` is of size 2^num_qubits, returns `matrix`.
+            Else it returns the block matrix (I = identity)
+            [[ embed_padding * I,    0    ],
+            [      0           , `matrix`]]
+
+        Raises:
+            ValueError: If embed_location is neither "upper" nor "lower".
+            ValueError: If the passed matrix does not fit into the space spanned by num_qubits.
+        """
+        full_dim = 1 << num_qubits
+        subs_dim = matrix.shape[0]
+
+        dim_diff = full_dim - subs_dim
+        if dim_diff == 0:
+            full_matrix = matrix
+
+        elif dim_diff > 0:
+            if embed_location == "lower":
+                full_matrix = np.zeros((full_dim, full_dim), dtype=complex)
+                full_matrix[:dim_diff, :dim_diff] = np.eye(dim_diff) * embed_padding
+                full_matrix[dim_diff:, dim_diff:] = matrix
```

Comment:
This seems to me like a lot of copying will take place, however, I am not 100% certain on the internals of numpy.
That said, I believe there may exist an easier implementation using `np.block` (see the first example here: https://numpy.org/doc/stable/reference/generated/numpy.block.html)